opensrp-chw/src/main/java/org/smartregister/chw/activity/ChildProfileActivity.java

```diff
@@ -67,6 +85,26 @@ protected void onCreation() {
         }
         notificationAndReferralRecyclerView.setAdapter(notificationListAdapter);
         notificationListAdapter.setOnClickListener(this);
+
+        if (getIntent().hasExtra(context().getStringResource(R.string.fhir_bundle))) {
+            createCarePlanEvent(getIntent().getStringExtra(context().getStringResource(R.string.fhir_bundle)));
+        }
+    }
+
+    private void createCarePlanEvent(String encodedBundle) {
```

Comment:
This should be executed on a background thread and probably in the presenter as activity should not have business logic

Reply:
Done